src/GitHub.App/Services/PullRequestService.cs

```diff
@@ -316,11 +316,6 @@ public IObservable<Unit> SwitchToBranch(ILocalRepositoryModel repository, IPullR
                         pullRequest.Head.Sha,
                         pullRequest.Base.Ref,
                         pullRequest.Head.Ref);
-
-                    if (sha == null)
-                    {
-                        throw new NotFoundException($"Couldn't find merge base between {pullRequest.Base.Sha} and {pullRequest.Head.Sha}.");
```

Comment:
This exception is now thrown from GetPullRequestMergeBase (with more info about what went wrong).

Reply:
What happens if an exception is thrown from a `Observable.Defer(async () =>`?